// @vertx/core/options.js
/*
 * Copyright 2020 ES4X
 *
 * ES4X licenses this file to you under the Apache License, version 2.0
 * (the "License"); you may not use this file except in compliance with the
 * License.  You may obtain a copy of the License at:
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

/// <reference types="@vertx/core/options" />

/**
 * @typedef { import("es4x") } Java
 */
module.exports = {
  AddressResolverOptions: Java.type('io.vertx.core.dns.AddressResolverOptions'),
  Argument: Java.type('io.vertx.core.cli.Argument'),
  ClientOptionsBase: Java.type('io.vertx.core.net.ClientOptionsBase'),
  CopyOptions: Java.type('io.vertx.core.file.CopyOptions'),
  DatagramSocketOptions: Java.type('io.vertx.core.datagram.DatagramSocketOptions'),
  DeliveryOptions: Java.type('io.vertx.core.eventbus.DeliveryOptions'),
  DeploymentOptions: Java.type('io.vertx.core.DeploymentOptions'),
  DnsClientOptions: Java.type('io.vertx.core.dns.DnsClientOptions'),
  EventBusOptions: Java.type('io.vertx.core.eventbus.EventBusOptions'),
  FileSystemOptions: Java.type('io.vertx.core.file.FileSystemOptions'),
  GoAway: Java.type('io.vertx.core.http.GoAway'),
  Http2Settings: Java.type('io.vertx.core.http.Http2Settings'),
  HttpClientOptions: Java.type('io.vertx.core.http.HttpClientOptions'),
  HttpServerOptions: Java.type('io.vertx.core.http.HttpServerOptions'),
  JdkSSLEngineOptions: Java.type('io.vertx.core.net.JdkSSLEngineOptions'),
  JksOptions: Java.type('io.vertx.core.net.JksOptions'),
  MetricsOptions: Java.type('io.vertx.core.metrics.MetricsOptions'),
  NetClientOptions: Java.type('io.vertx.core.net.NetClientOptions'),
  NetServerOptions: Java.type('io.vertx.core.net.NetServerOptions'),
  NetworkOptions: Java.type('io.vertx.core.net.NetworkOptions'),
  OpenOptions: Java.type('io.vertx.core.file.OpenOptions'),
  OpenSSLEngineOptions: Java.type('io.vertx.core.net.OpenSSLEngineOptions'),
  Option: Java.type('io.vertx.core.cli.Option'),
  PemKeyCertOptions: Java.type('io.vertx.core.net.PemKeyCertOptions'),
  PemTrustOptions: Java.type('io.vertx.core.net.PemTrustOptions'),
  PfxOptions: Java.type('io.vertx.core.net.PfxOptions'),
  ProxyOptions: Java.type('io.vertx.core.net.ProxyOptions'),
  RequestOptions: Java.type('io.vertx.core.http.RequestOptions'),
  StreamPriority: Java.type('io.vertx.core.http.StreamPriority'),
  TCPSSLOptions: Java.type('io.vertx.core.net.TCPSSLOptions'),
  VertxOptions: Java.type('io.vertx.core.VertxOptions'),
  WebSocketConnectOptions: Java.type('io.vertx.core.http.WebSocketConnectOptions')
};
